File: src/pre/colorfilter/MetalTexFilter.ts
class MetalTexFilter extends MetalColorFilter {

    private texFilter: TexColorFilter;

    constructor(color: string, shinyness: number, brightness: number, hueshift: number, bitmap: android.graphics.Bitmap){
        super(color, shinyness, brightness, hueshift);
        this.texFilter = new TexColorFilter(bitmap);
    }

    preProcess(data: number[]): void {
        const bitmap = this.texFilter.applyTo(this.baseTex);
        for(let i = 0; i < data.length; i++){
            data[i] = bitmap.getPixel(ColorFilter.getX(i), ColorFilter.getY(i));
        }
    }

}

File: texture-generator/filters/metal_tex_filter.py
from . import MetalColorFilter, TexColorFilter, RGBA


class MetalTexFilter(MetalColorFilter):

    def __init__(self, color_hex: str, shinyness: float, brightness: float, hueshift: float, tex_path: str):
        super().__init__(color_hex, shinyness, brightness, hueshift)
        self.tex_filter = TexColorFilter(tex_path)
    
    def pre_process(self, data: list[RGBA]):
        data2 = self.tex_filter.apply(data)
        for i in range(len(data)) :
            data[i] = data2[i]
